cgaal-engine/src/algorithms/certain_zero/search_strategy/composite.rs

```diff
@@ -0,0 +1,209 @@
+    Dfs(DepthFirstSearch<AtlVertex>),
+    Rdfs(RandomDepthFirstSearch<AtlVertex>),
+    Los(LinearOptimizeSearch),
+    Lps(LinearProgrammingSearch),
+    Lrs(LinearRepresentativeSearch),
+    Dhs(DependencyHeuristicSearch<AtlVertex>),
+    Ihs(InstabilityHeuristicSearch),
+}
+
+impl SearchStrategy<AtlVertex> for CompositeSearchStrategyInstance {
+    fn next(&mut self) -> Option<Edge<AtlVertex>> {
+        use CompositeSearchStrategyInstance as CssInst;
+        match self {
+            CssInst::Bfs(ss) => ss.next(),
+            CssInst::Dfs(ss) => ss.next(),
+            CssInst::Rdfs(ss) => ss.next(),
+            CssInst::Los(ss) => ss.next(),
+            CssInst::Lps(ss) => ss.next(),
+            CssInst::Lrs(ss) => ss.next(),
+            CssInst::Dhs(ss) => ss.next(),
+            CssInst::Ihs(ss) => ss.next(),
+        }
+    }
+
+    fn queue_new_edges(&mut self, edges: Vec<Edge<AtlVertex>>) {
+        use CompositeSearchStrategyInstance as CssInst;
+        match self {
+            CssInst::Bfs(ss) => ss.queue_new_edges(edges),
+            CssInst::Dfs(ss) => ss.queue_new_edges(edges),
+            CssInst::Rdfs(ss) => ss.queue_new_edges(edges),
+            CssInst::Los(ss) => ss.queue_new_edges(edges),
+            CssInst::Lps(ss) => ss.queue_new_edges(edges),
+            CssInst::Lrs(ss) => ss.queue_new_edges(edges),
+            CssInst::Dhs(ss) => ss.queue_new_edges(edges),
+            CssInst::Ihs(ss) => ss.queue_new_edges(edges),
+        }
+    }
+}
+
+/// The [CompositeSearchStrategyBuilder] sequentially produces search strategies from a vec of
+/// selected search strategies and thus makes it possible to use of different search
+/// strategies across different worker. A search strategy can be selected multiple times.
+///
+/// E.g. if the selected strategies are BFS, BFS, RDFS, then every third worker will use the
+/// RDFS strategy, while the remaining workers will use the BFS strategy.
+pub struct CompositeSearchStrategyBuilder {
+    pub game: IntermediateLcgs,
+    pub strategies: Vec<CompositeSearchStrategyOption>,
+    pub index: RefCell<usize>,
+}
+
+impl CompositeSearchStrategyBuilder {
+    pub fn new(
+        game: IntermediateLcgs,
+        strategies: Vec<CompositeSearchStrategyOption>,
+    ) -> CompositeSearchStrategyBuilder {
+        assert!(
+            !strategies.is_empty(),
+            "Composite search strategy must have at least one component search strategy"
+        );
+        CompositeSearchStrategyBuilder {
+            game,
+            strategies,
+            index: RefCell::new(0),
+        }
+    }
+
+    /// Create a new [CompositeSearchStrategyBuilder] employing each search strategy evenly.
+    pub fn one_of_each(game: IntermediateLcgs) -> CompositeSearchStrategyBuilder {
+        use CompositeSearchStrategyOption::*;
+        // General/best strategies first in case the user is using few workers
+        Self::new(game, vec![Dhs, Rdfs, Ihs, Bfs, Lrs, Dfs, Los, Lps])
```

Comment:
Spicy but sane choice. We should be aware that this default is to be changed, if we discover speedups elsewhere.